### ISSUE TITLE
Let's use spdx-tools for SBOM generation

### ISSUE THREAD

**Issue by carmenbianca** [fsfe/reuse-tool#671]
We can get this for free instead of doing the string manipulation we do now.

**Comment by seabass-labrax** [comment]
The SPDX Python libraries have been getting a lot of attention recently, with lots of refactoring and feature updates. If there anything in particular that you would need added for this, please let me know and I'll see what I can do! :smiley:

**Comment by nicorikken** [comment]
Agreed. I've looked a bit at the [GitHub project](https://github.com/spdx/tools-python) and it seems a thorough library. It is promising that there is progress on the hard nuts to crack https://github.com/spdx/tools-python/issues/10 It will also better the ecosystem overall if we collaborate in this generic library instead of doing our own thing.

**Comment by meretp** [comment]
Hi @nicorikken, @carmenbianca! Very cool, that you want to use the `tools-python`. Regarding your second concern, nico, we already have an open [issue](https://github.com/spdx/tools-python/issues/454) for that and want to reduce the dependencies, the hint on `Click` is already helpful! Probably we will make these changes in the [refactored-branch](https://github.com/spdx/tools-python/tree/refactor-python-tools) where we are working on a refactored version of the tools that has separated layers for parsing, validating and reading files. We want to bring this branch to main soon and then keep the current, stable version next to it. I would advise you to take a look at the refactored version, which has breaking changes to the current one, and consider using it in the future. The current version on main will get after the switch only bug fixes, but tend to get no new features in the future.

**Comment by armintaenzertng** [comment]
FYI: the refactored version of `spdx/tools-python` can now be found on the [main branch](https://github.com/spdx/tools-python).

**Comment by mxmehl** [comment]
Related to #394
